### DISCUSSION TITLE
Downsize image to smarpthone screen width: how to achieve same quality as browser does?

### DISCUSSION THREAD

**Discussion by alex-rsk** [ImageMagick/ImageMagick#4560] [Help]
Hello.
Brief history: last time I was need to shrink quadratic images to size of mobile screen, for speeding up our internet store on mobile platforms.
Firstly I made resize with standard  GD library (we are using PHP), but the quality of resized images was just horrible, as well as file size wasn't reduced much.

So, I decided to resize with Image Magick (version 6.9, Linux). Results were better, but not enough.

Images were either too blurry, or vice versa - too sharp. When I was attempting to apply different filters,  there was ringing or aliasing or blur or color distortion etc, etc.  Anyway, losing quality was noticeable. While any contemporary browser always scales  down big picture (1080х1080)  with **excellent quality**. How it does it? I don't think they know some secret formulae, this is just technique. 
But before I dig into chromium sources, I dare to ask the community.

I tried several approaches:
1. Resizing with every existing filter 
2. Unsharp after resize
3. Tried to scale down not to exact size but width divided to 2 (anyway browser did the rest of work).

My attempts are all summarized in this page:
http://45.135.134.109/gray_woman/375
http://45.135.134.109/gray_pants/375
http://45.135.134.109/big_rtrend/375

You can play with settings and estimate the result.
The code is here: https://github.com/alex-rsk/resize_with_im
Any advices would be appreciated.

**Comment by snibgo** [comment]
You seem to have an input JPEG that is 1080x1080. You resize that with IM to 540x540, JPEG. Then you display that 540x540 on a web page with the image resized (by the browser) to 375x375. Is that correct?

Two resizes are worse than one. I suggest you should resize once only.

Saving as JPEG is generally lossy. Your input 1080x1080 JPEGs show some artifacts at the edges. That will not help the image quality. I suggest always starting from the best quality sources that have not been through JPEG compression.

**Comment by alex-rsk** [comment]
>> You seem to have an input JPEG that is 1080x1080. You resize that with IM to 540x540, JPEG. Then you display that 540x540 on a web page with the image resized (by the browser) to 375x375. Is that correct?

Actually not. 
If you proceed to link  http://45.135.134.109/gray_woman/375, it will resize 1080 to 375.
But if you want to compare how it looks with TWO resizes, please set checkbox "x/2" and press "without sharpness".
Suprisingly, the second one looks better, despite it makes 2 resizes (first, with IM from 1080 to 540, and then browser resizes 540 to 375 to fit screen).
Thus, browser resize works **better**, than IM single resize.

>>Saving as JPEG is generally lossy. Your input 1080x1080 JPEGs show some artifacts at the edges. That will not help the image quality. I suggest always starting from the best quality sources that have not been through JPEG compression.

All our input is from Instagram account, which displays all in JPEG, sadly.
